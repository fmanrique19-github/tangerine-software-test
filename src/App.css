.App {
  text-align: center;
}

.app-logo {
  width: 100%;
  max-width: 367px;
  margin: auto;
  display: block;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #09d3ac;
}

.tangerine-login {
  margin: 40px auto;
  display: block;
  width: 100%;
}

.tangerine-container {
  margin-top: 140px;
  display: block;
  width: 100%;
}

.tangerine-footer {
  clear: both;
}

.tangerine-fixed {
    position: fixed;
    right: 0;
}

.photo-mobile-close {
  display: none;
}

.padding-t {
  padding-top: 5px;
  display: block;
}

@media (max-width: 750px) {
  .hide-albums-mobile {
    display: none;
  }

  .photo-mobile-close {
    display: block;
    margin-bottom: 20px;
    color: #d50000;
    cursor: pointer;
  }

  .tangerine-container {
    margin-top: 100px;
  }

  .scrollbar {
    overflow: hidden !important;
    height: 100% !important;
  }

  .tangerine-fixed {
    position: relative;
  }
}

.scrollbar {
  overflow: scroll;
  height: 550px;
  overflow-x: hidden;
  scroll-behavior: smooth;
}

.scrollbar::-webkit-scrollbar {
  width: 6px;
  height: 6px;
}

.scrollbar::-webkit-scrollbar-thumb {
  border-radius: 5px;
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.1);
  background: rgba(0, 0, 0, 0.2);
}

/* custom colors  */
.scrollbar-primary::-webkit-scrollbar {
  background-color: #F5F5F5;
}

.scrollbar-primary::-webkit-scrollbar-thumb {
  border-radius: 10px;
  -webkit-box-shadow: inset 0 0 6px rgba(0, 0, 0, 0.1);
  background-color: #4285F4;
}