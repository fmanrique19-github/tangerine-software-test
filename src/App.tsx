import './App.css';
import React, { Suspense } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Login from './components/Login';
import Home from  './components/Home';
import Author from  './components/Author';


export interface IProps {}

export interface IState {}

class App extends React.Component<IProps, IState> {

  public componentDidMount() {
    
    this.setLogin();
  }

  public setLogin() {

    const isLogin = localStorage.getItem('isLogin');

    const pathname = window.location.pathname;

    if((isLogin === 'false' && pathname !== '/login' && pathname !== '/author') || (isLogin === 'false' && pathname === '/author')) {
        window.location.pathname = '/login';
    } else if(isLogin === 'true' && pathname !== '/home' && pathname !== '/author') {
      window.location.pathname = '/home';
    }
  }


  public render() {
      return (
        <Router>
          <Suspense fallback={<div></div>}>
            <Switch>
              <Route exact path='/' key='0'><Redirect to="/login" /></Route>
              <Route exact path='/login' component={Login} key='1' />
              <Route exact path='/home' component={Home} key='2' />
              <Route exact path='/author' component={Author} key='3' />
            </Switch>
          </Suspense>
        </Router>
      );
  }
}

export default App;
