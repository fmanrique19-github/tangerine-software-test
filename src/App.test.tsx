import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import App from './App';
import mockStore from './tests/helpers/mockStore';

it('renders without crashing', () => {
  
  const props = {
    user: {
        isLogin: false,
    },
    albums: null,
    photos: null,
  };

  const tree = renderer
      .create(<Provider store={mockStore(props)}><App {...props} /></Provider>)
      .toJSON();
  expect(tree).toMatchSnapshot();

});
