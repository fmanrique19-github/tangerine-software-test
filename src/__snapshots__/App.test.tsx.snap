// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="container tangerine-login"
  data-test="container"
>
  <div
    className="row"
    data-test="row"
  >
    <div
      className="col"
      data-test="col"
    >
      <div
        className="card"
        data-test="card"
      >
        <div
          className="header pt-3 warning-color-dark lighten-2"
        >
          <h3
            className="text-white mt-3 mb-4 pb-1 mx-5"
          >
            Log in
          </h3>
        </div>
        <div
          className="card-body"
          data-test="card-body"
        >
          <img
            alt="Tangerine Software"
            className="app-logo"
            src="https://i.ibb.co/smryvqs/tangerine-logo.jpg"
          />
          <div
            className="grey-text"
          >
            <div
              className="mr-2 float-left"
            >
              <svg
                fill="none"
                height="32"
                id="i-user"
                stroke="currentcolor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 11 C22 16 19 20 16 20 13 20 10 16 10 11 10 6 12 3 16 3 20 3 22 6 22 11 Z M4 30 L28 30 C28 21 22 20 16 20 10 20 4 21 4 30 Z"
                />
              </svg>
            </div>
            <div
              className="md-form"
            >
              <i
                className="fa fa-none prefix"
                data-test="fa"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
              <input
                className="form-control"
                data-test="input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                required={true}
                type="text"
                value=""
              />
              <label
                className=""
                data-error="eeeded"
                data-success=""
                onClick={[Function]}
              >
                Username
              </label>
            </div>
            <div
              className="mr-2 float-left"
            >
              <svg
                fill="none"
                height="32"
                id="i-lock"
                stroke="currentcolor"
                strokeLinecap="round"
                strokeLinejoin="round"
                strokeWidth="2"
                viewBox="0 0 32 32"
                width="32"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5 15 L5 30 27 30 27 15 Z M9 15 C9 9 9 5 16 5 23 5 23 9 23 15 M16 20 L16 23"
                />
                <circle
                  cx="16"
                  cy="24"
                  r="1"
                />
              </svg>
            </div>
            <div
              className="md-form"
            >
              <i
                className="fa fa-none prefix"
                data-test="fa"
                onClick={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              />
              <input
                className="form-control"
                data-test="input"
                disabled={false}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                onInput={[Function]}
                type="password"
                value=""
              />
              <label
                className=""
                data-error=""
                data-success=""
                onClick={[Function]}
              >
                Password
              </label>
            </div>
            <a
              className="d-flex  justify-content-end blue-text ml-1"
              href="/forgot-password"
              onClick={[Function]}
            >
              Forgot password?
            </a>
          </div>
          <div
            className="text-center py-4 mt-3"
          >
            <button
              className="btn-primary btn Ripple-parent"
              data-test="button"
              onClick={[Function]}
              onMouseUp={[Function]}
              onTouchStart={[Function]}
              type="button"
            >
              <h4
                className="float-left mt-1"
              >
                Send
              </h4>
              <div
                className="ml-1 float-left"
              >
                <svg
                  fill="none"
                  height="32"
                  id="i-signin"
                  stroke="currentcolor"
                  strokeLinecap="round"
                  strokeLinejoin="round"
                  strokeWidth="2"
                  viewBox="0 0 32 32"
                  width="32"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M3 16 L23 16 M15 8 L23 16 15 24 M21 4 L29 4 29 28 21 28"
                  />
                </svg>
              </div>
              <div
                className="Ripple "
                data-test="waves"
                style={
                  Object {
                    "height": "0px",
                    "left": "0px",
                    "top": "0px",
                    "width": "0px",
                  }
                }
              />
            </button>
          </div>
        </div>
        <div
          className="modal-footer mx-5 pt-3 mb-1"
          data-test="modal-footer"
        >
          <a
            className="blue-text ml-1"
            href="/create-account"
            onClick={[Function]}
          >
            Create account
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
